crates/bevy_animation/src/animatable.rs

```diff
@@ -0,0 +1,162 @@
+use crate::util;
+use bevy_ecs::world::World;
+use bevy_math::*;
+use bevy_reflect::Reflect;
+use bevy_transform::prelude::Transform;
+use bevy_utils::FloatOrd;
+
+/// An individual input for [`Animatable::blend`].
+pub struct BlendInput<T> {
+    /// The individual item's weight. This may not be bound to the range `[0.0, 1.0]`.
+    pub weight: f32,
+    /// The input value to be blended.
+    pub value: T,
+    /// Whether or not to additively blend this input into the final result.
+    pub additive: bool,
+}
+
+/// An animatable value type.
+pub trait Animatable: Reflect + Sized + Send + Sync + 'static {
+    /// Interpolates between `a` and `b` with  a interpolation factor of `time`.
+    ///
+    /// The `time` parameter here may not be clamped to the range `[0.0, 1.0]`.
```

Comment:
What does this comment mean? How is an implementor supposed to understand interpolating with `t=1.5` (which sounds like an extrapolation rather)? Are we sure this is even possible in the general case? Because this seems to strongly imply `lerp` here / a linear transform.